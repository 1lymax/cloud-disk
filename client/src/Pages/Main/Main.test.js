import {screen, waitFor} from '@testing-library/react';
import {renderTestApp} from "../../tests/helpers/renderTestApp";


describe('Main page', () => {
	test('match snapshot', async () => {
		renderTestApp(null, {
			route: '/'
		})

		expect(await screen.getByTestId('main-page')).toBeInTheDocument()
		const skeletonElem = await screen.getAllByTestId('skeleton-elem')
		expect(skeletonElem.length).toBe(3)
		await waitFor(() => {
			const countElem = screen.getAllByTestId('count-elem')
			expect(countElem[0]).toContainHTML('6')
			expect(countElem[1]).toContainHTML('4')
			expect(countElem[2]).toContainHTML('2')
			expect(screen.getByTestId('main-page')).toMatchSnapshot()
		})
	})
});
