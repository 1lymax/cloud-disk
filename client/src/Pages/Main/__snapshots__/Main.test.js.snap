// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page match snapshot 1`] = `
<div
  class="main__wrapper"
  data-testid="main-page"
>
  <div
    class="main"
  >
    <div
      class="main__title"
    >
      Our service is
    </div>
    <div
      class="main__users main__item__wrapper"
    >
      <div
        class="main__item__header"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-xhjocd-MuiSvgIcon-root"
          data-testid="PeopleAltIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.67 13.13C18.04 14.06 19 15.32 19 17v3h4v-3c0-2.18-3.57-3.47-6.33-3.87z"
            fill-rule="evenodd"
          />
          <circle
            cx="9"
            cy="8"
            fill-rule="evenodd"
            r="4"
          />
          <path
            d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4c-.47 0-.91.1-1.33.24C14.5 5.27 15 6.58 15 8s-.5 2.73-1.33 3.76c.42.14.86.24 1.33.24zm-6 1c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
            fill-rule="evenodd"
          />
        </svg>
        <div
          data-testid="count-elem"
        >
          6
        </div>
      </div>
      <div
        class="main__item__description"
      >
        registered users
      </div>
    </div>
    <div
      class="main__space main__item__wrapper"
    >
      <div
        class="main__item__header"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-xhjocd-MuiSvgIcon-root"
          data-testid="CloudDoneIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM10 17l-3.5-3.5 1.41-1.41L10 14.17 15.18 9l1.41 1.41L10 17z"
          />
        </svg>
        <div
          data-testid="count-elem"
        >
          4
        </div>
         B
      </div>
      <div
        class="main__item__description"
      >
        used space
      </div>
    </div>
    <div
      class="main__files main__item__wrapper"
    >
      <div
        class="main__item__header"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-xhjocd-MuiSvgIcon-root"
          data-testid="SaveAsIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 12.4V7l-4-4H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h7.4l8.6-8.6zM15 15c0 1.66-1.34 3-3 3s-3-1.34-3-3 1.34-3 3-3 3 1.34 3 3zM6 6h9v4H6V6zm13.99 10.25 1.77 1.77L16.77 23H15v-1.77l4.99-4.98zm3.26.26-.85.85-1.77-1.77.85-.85c.2-.2.51-.2.71 0l1.06 1.06c.2.2.2.52 0 .71z"
          />
        </svg>
        <div
          data-testid="count-elem"
        >
          2
        </div>
      </div>
      <div
        class="main__item__description"
      >
        uploaded files
      </div>
    </div>
    <div
      class="main__try main__item__wrapper"
    >
      <div
        class="main__try__link"
      >
        try it for free.
      </div>
    </div>
  </div>
</div>
`;
